### PR TITLE
Nucleus terraform

### PR DESCRIPTION
**Pull request by alexdunnjpl**
## 🗒️ Summary

Terraform script simplified with external modules and added integration with Nucleus/Airflow to trigger and monitor the sweeper instead of event bridge previously.

### 🤖 AI Assistance Disclosure
- [ ] No AI assistance used
- [ ] AI used for light assistance (e.g., suggestions, refactoring, documentation help, minor edits)
- [ ] AI used for moderate content generation (AI generated some code or logic, but the developer authored or heavily revised the majority)
- [ ] AI generated substantial portions of this code

Estimated % of code influenced by AI: ___ %

## ⚙️ Test Data and/or Report

Validated with registry integration tests run on the AWS deployment.

## 🤓 Reviewer Checklist

*Reviewers: Please verify the following before approving this pull request.*

### Documentation and PR Content
- [ ] **Documentation:** README, Wiki, or inline documentation (Sphinx, Javadoc, Docstrings) have been updated to reflect these changes.
- [ ] **Issue Traceability:** The PR is linked to a valid GitHub Issue
- [ ] **PR Title:** The PR title is "user-friendly" clearly identifying what is being fixed or the new feature being added, that if you saw it in the Release Notes for a tool, you would be able to get the gist of what was done.

### Security & Quality
- [ ] **SonarCloud:** Confirmed no new High or Critical security findings.
- [ ] **Secrets Detection:** Verified that the Secrets Detection scan passed and no sensitive information (keys, tokens, PII) is exposed.
- [ ] **Code Quality:** Code follows organization style guidelines and best practices for the specific language (e.g., PEP 8, Google Java Style).

### Testing & Validation
- [ ] **Test Accuracy:** Verified that test data is accurate, representative of real-world PDS4 scenarios, and sufficient for the logic being tested.
- [ ] **Coverage:** Automated tests cover new logic and edge cases.
- [ ] **Local Verification:** (If applicable) Successfully built and ran the changes in a local or staging environment.

### Maintenance
- [ ] **Backward Compatibility:** Confirmed that these changes do not break existing downstream dependencies or API contracts (or that breaking changes are clearly documented).

### CODE REVIEW

File: terraform/variables.tf [hunk 1]
```diff
@@ -1,25 +1,22 @@
-variable "aws_region" {
-  description = "AWS Region"
-  type        = string
-  default     = "us-west-2"
-}
+venue              = "prod"
+aoss_endpoint      = "https://..."
+aoss_collection_id = "..."
+image_uri          = "123456789012.dkr.ecr.us-west-2.amazonaws.com/pds-registry-sweepers:latest"
```
Comment (tloubrieu-jpl) [inline terraform/variables.tf]: Note that we should be able to use a cached version of the docker image if it is published on github docker artifactory, as done for registry api, see https://github.com/NASA-PDS/registry-api/blob/c910a5c810eb8ab1ee1c4a188c497ce50b3d1998/.github/workflows/unstable-cicd.yaml#L96

That will spare us to manually push on ECR our images.

That does not require any changes in the current PR.

File: terraform/variables.tf [hunk 2]
```diff
@@ -1,25 +1,22 @@
-variable "aws_region" {
```
Comment (tloubrieu-jpl) [inline terraform/variables.tf:1]: Sorry I guess I was unclear at some point, the usage is to have the variable declaration (variable "aws_region" { ...) in this file.
The actual value proposed as example would be in a `terraform.tvfars.example` so that the user have their own terraform.tfvars locally inspired by the example.

**Review by tloubrieu-jpl — CHANGES_REQUESTED** [review]

**Comment by tloubrieu-jpl** [comment]
I updated the terraform code to make it consistent with the other repo I am aware of. I also updated the resource tags so that the deployment works in dev by complying with our tagging policy.

File: terraform/variables.tf (hunk 2 above)
Comment (alexdunnjpl) [reply terraform/variables.tf:1]: Isn't this value autoresolvable from the executing context via [this datasource](https://github.com/NASA-PDS/registry-sweepers/pull/219/changes/BASE..190e820dd4a624e4bc8f55156e0638779cf616dc#diff-2e617c7870fa918457b1eee1c7d67ba82f19d043ae7b7918db26873e18793028R8)?

I might not be understanding the need to provide it explicitly as a tfvar

File: terraform/variables.tf (hunk 1 above)
Comment (alexdunnjpl) [reply terraform/variables.tf]: @tloubrieu-jpl roger that.

Just to check, is there any potential for delays or invalidation issues with this setup?  Just want to be aware so that if something doesn't seem to be working as expected right after an update, I know whether to check that the docker image isn't stale and what to do about it if it is.

Comment (tloubrieu-jpl) [reply terraform/variables.tf]: I believe the performances are not as good as anytime the task starts it first needs to pull it from docker hub. So in production we will need to switch to an ECR local image (using Pull Through Cache if we can make that work so that we don't need to anually push every update to ECR).

File: terraform/variables.tf (hunk 2 above)
Comment (tloubrieu-jpl) [reply terraform/variables.tf:1]: You are right about the aws_region being auto-resolvable, somehow we have this explicit variable in many scripts, however I was not refering to aws_region specifically but all the variables declarations are usually in their own variables.tf file. I made that change, so we are good with that.

**Review by tloubrieu-jpl — APPROVED** [review]
Reviewed and tested
